tests/route/test_default_route.py

```diff
@@ -13,7 +16,88 @@
+    """
+    Get the information for upstream neighbors present in the testbed
+    
+    returns dict: {"upstream_neigh_name" : (ipv4_intf_ip, ipv6_intf_ip)} 
+    """
+
+    def get_upstream_neigh_type(topo):
+        if 't0' in topo or 'dualtor' in topo:
+            return 't1'
+        elif 't1' in topo:
+            return 't2'
+        elif 't2' in topo:
+            return 't3'
+        else:
+            return None
+
+    upstream_neighbors = {}
+    neigh_type = get_upstream_neigh_type(tb['topo']['name'])
+    logging.info("testbed topo {} upstream neigh type {}".format(
+        tb['topo']['name'], neigh_type))
+
+    topo_cfg_facts = tb['topo']['properties'].get('configuration', None)
+    if topo_cfg_facts is None:
+        return upstream_neighbors
+
+    for neigh_name, neigh_cfg in topo_cfg_facts.iteritems():
+        if neigh_type not in neigh_name.lower():
+            continue
+        interfaces = neigh_cfg.get('interfaces', {})
+        ipv4_addr = None
+        ipv6_addr = None
+        for intf, intf_cfg in interfaces.iteritems():
+            if 'Port-Channel' in intf:
+                if 'ipv4' in intf_cfg:
+                    ipv4_addr = interfaces[intf]['ipv4'].split('/')[0]
+                if 'ipv6' in intf_cfg:
+                    ipv6_addr = interfaces[intf]['ipv6'].split('/')[0]
+            elif 'Ethernet' in intf:
+                if 'ipv4' in intf_cfg:
+                    ipv4_addr = interfaces[intf]['ipv4']
+                if 'ipv6' in intf_cfg:
+                    ipv6_addr = interfaces[intf]['ipv6']
+            else:
+                continue
+
+        upstream_neighbors.update({neigh_name: (ipv4_addr, ipv6_addr)})
+    return upstream_neighbors
+
+
+def verify_default_route_in_app_db(asichost, tbinfo, af):
+    """
+    Verify the nexthops for the default routes match the ip interfaces
+    configured on the peer device 
+    """
+    default_route = asichost.get_default_route_from_app_db(af)
+    pytest_assert(default_route, "default route not present in APP_DB")
+    logging.info("default route from app db {}".format(default_route))
+    # There is only one default route in app_db
+    key = default_route.keys()[0]
+
+    nexthop_list = default_route[key].get('value', {}).get('nexthop', None)
+    pytest_assert(nexthop_list is not None, "Default route has not nexthops")
+    logging.info("nexthop list in app_db {}".format(nexthop_list) )
+    nexthops = set(nexthop_list.split(','))
+    upstream_neigh = get_upstream_neigh(tbinfo)
+    pytest_assert(upstream_neigh is not None, "No upstream neighbors in the testbed")
+
+    if af == 'ipv4':
+        upstream_neigh_ip = set([upstream_neigh[neigh][0]  for neigh in upstream_neigh])
+    elif af == 'ipv6':
+        upstream_neigh_ip= set([upstream_neigh[neigh][1]  for neigh in upstream_neigh])
+
+    logging.info("peer intf ip from tb {}".format(upstream_neigh_ip))
+    pytest_assert(len(nexthops) == len(upstream_neigh_ip), \
```

Comment:
Hi Arvind, this won't work for multi-asic if there are 2 backend asics that both connect to upstream neighbors.
While nexthops are coming from one asic only.